### ISSUE TITLE
Enchant 2 not detected

### ISSUE THREAD

**Issue by Arfrever** [fcitx/fcitx#356]
After upgrade of Enchant from version 1.6.* to 2.1.0, build system of Fcitx does not detect Enchant:
```
-- Checking for module 'enchant'
--   Found enchant, version 2.1.0
-- Performing Test ENCHANT_HAVE_WERROR
-- Performing Test ENCHANT_HAVE_WERROR - Success
-- Performing Test ENCHANT_API_COMPATIBLE
-- Performing Test ENCHANT_API_COMPATIBLE - Failed
-- Could NOT find Enchant (missing: ENCHANT_API_COMPATIBLE)
```

**Comment by wengxt** [comment]
fixed by 6b4a459

**Comment by Arfrever** [comment]
I see that e.g. `enchant_dict_add_to_personal` is used in fcitx5, so that repository also needs fixing.
